core/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -53,4 +118,97 @@ public static void main(String[] args) throws StartupError {
     static void close(String[] args) throws IOException {
         Bootstrap.stop();
     }
+
+    @Argument(handler = SubCommandHandler.class, metaVar = "command")
+    @SubCommands({
+        @SubCommand(name = "start", impl = StartCommand.class),
+        @SubCommand(name = "version", impl = VersionCommand.class)
+    })
+    Command command;
+
+    static abstract class Command {
+
+        @Option(name = "-h", aliases = "--help", usage = "print this message")
```

Comment:
I agree we should just support `--version`. And I don't see why we need these "commands" here, the logic should be simple: ```if (--version) printVersionAndExit();```